test/esinstall.api.test.js

```diff
@@ -0,0 +1,221 @@
+const {findExportMapEntry, findManifestEntry, explodeExportMap} = require('../esinstall/lib');
+
+describe('ESInstall API', () => {
+  describe('findExportMapEntry', () => {
+    it('returns a string value', () => {
+      expect(findExportMapEntry('foo.js')).toBe('foo.js');
+    });
+
+    it('prefers browser', () => {
+      expect(
+        findExportMapEntry({
+          browser: 'browser.js',
+          import: 'import.js',
+          default: 'default.js',
+          require: 'require.js',
+        }),
+      ).toBe('browser.js');
+    });
+
+    it('picks import', () => {
+      expect(
+        findExportMapEntry({
+          import: 'import.js',
+          default: 'default.js',
+          require: 'require.js',
+        }),
+      ).toBe('import.js');
+    });
+
+    it('picks default', () => {
+      expect(
+        findExportMapEntry({
+          default: 'default.js',
+          require: 'require.js',
+        }),
+      ).toBe('default.js');
+    });
+
+    it('picks require', () => {
+      expect(
+        findExportMapEntry({
+          require: 'require.js',
+          other: 'other.js',
+        }),
+      ).toBe('require.js');
+    });
+
+    it('takes conditions', () => {
+      expect(
+        findExportMapEntry(
+          {
+            development: {
+              browser: 'error.js',
+            },
+            production: {
+              import: 'prod.js',
+            },
+          },
+          ['production'],
+        ),
+      ).toBe('prod.js');
+    });
+  });
+
+  describe('findManifestEntry', () => {
+    it('gets "exports"', () => {
+      let entry = findManifestEntry({exports: 'esm.js', main: 'main.js'});
+      expect(entry).toBe('esm.js');
+    });
+
+    it('gets "exports[.]"', () => {
+      let entry = findManifestEntry({
+        exports: {
+          '.': 'esm.js',
+        },
+        main: 'main.js',
+      });
+      expect(entry).toBe('esm.js');
+    });
+
+    it('gets "main"', () => {
+      let entry = findManifestEntry({main: 'foo.js'});
+      expect(entry).toBe('foo.js');
+    });
+
+    it('gets "module"', () => {
+      let entry = findManifestEntry({main: 'error.js', module: 'foo.js'});
+      expect(entry).toBe('foo.js');
+    });
+
+    it('gets "browser:module"', () => {
+      let entry = findManifestEntry({main: 'error.js', 'browser:module': 'mod.js'});
+      expect(entry).toBe('mod.js');
+    });
+
+    it('gets "main:esnext"', () => {
+      let entry = findManifestEntry({main: 'error.js', 'main:esnext': 'main.js'});
+      expect(entry).toBe('main.js');
+    });
+
+    it('gets "jsnext:main"', () => {
```

Comment:
👍🏻  Great job keeping `main:esnext` and `jsnext:main` and all the old ones. This is really good for old packages